### PR TITLE
[iOS11] Update project settings and fix errors

### PR DESCRIPTION
**Pull request by Eke**
Hello!

I fixed some warnings in Xcode 9 and iOS 11. There are still some warnings in dependencies when running `pod lib lint --verbose`. This PR can unblock me with #653 

```
Testing with xcodebuild.
 -> Texture (2.6)
    - WARN  | [Texture/IGListKit] xcodebuild:  IGListKit/Source/IGListUpdatingDelegate.h:34:38: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/IGListKit] xcodebuild:  IGListKit/Source/IGListUpdatingDelegate.h:38:40: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/IGListKit] xcodebuild:  IGListKit/Source/IGListAdapter.m:941:18: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/IGListKit] xcodebuild:  IGListKit/Source/Internal/IGListBatchUpdates.h:26:64: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/IGListKit] xcodebuild:  IGListKit/Source/IGListAdapterUpdater.m:425:50: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/IGListKit] xcodebuild:  IGListKit/Source/IGListAdapterUpdater.m:55:26: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/IGListKit] xcodebuild:  IGListKit/Source/IGListAdapterUpdater.m:133:32: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/IGListKit] xcodebuild:  IGListKit/Source/IGListAdapterUpdater.m:181:24: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/IGListKit] xcodebuild:  {{PATH}}/Xcode/DerivedData/App-akhbasdbsjtjlsgxcsxwstypgecs/Build/Products/Release-iphonesimulator/IGListKit/IGListKit.framework/Headers/IGListUpdatingDelegate.h:34:38: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/IGListKit] xcodebuild:  {{PATH}}/Xcode/DerivedData/App-akhbasdbsjtjlsgxcsxwstypgecs/Build/Products/Release-iphonesimulator/IGListKit/IGListKit.framework/Headers/IGListUpdatingDelegate.h:38:40: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [Texture/Yoga] xcodebuild:  {{PATH}}/Texture/Source/ASDisplayNode+Beta.h:180:44: warning: 'UISemanticContentAttribute' is partial: introduced in iOS 9.0 [-Wunguarded-availability]
    - NOTE  | [Texture/Yoga] xcodebuild:  /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.1.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIView.h:121:28: note: 'UISemanticContentAttribute' has been explicitly marked partial here
    - NOTE  | [Texture/Yoga] xcodebuild:  /{{PATH}}/Texture/Source/ASDisplayNode+Beta.h:180:1: note: annotate 'semanticContentAttributeDidChange:' with an availability attribute to silence
    - WARN  | [Texture/Yoga] xcodebuild:  /{{PATH}}/Texture/Source/ASDisplayNode+Yoga.mm:95:44: warning: 'UISemanticContentAttribute' is partial: introduced in iOS 9.0 [-Wunguarded-availability]
    - NOTE  | [Texture/Yoga] xcodebuild:  /{{PATH}}/Texture/Source/ASDisplayNode+Yoga.mm:95:1: note: annotate 'semanticContentAttributeDidChange:' with an availability attribute to silence
    - WARN  | [Texture/Yoga] xcodebuild:  /{{PATH}}/Texture/Source/ASDisplayNode+Yoga.mm:99:13: warning: 'userInterfaceLayoutDirectionForSemanticContentAttribute:' is only available on iOS 9.0 or newer [-Wunguarded-availability]
    - NOTE  | [Texture/Yoga] xcodebuild:  /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.1.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIView.h:167:1: note: 'userInterfaceLayoutDirectionForSemanticContentAttribute:' has been explicitly marked partial here
    - NOTE  | [Texture/Yoga] xcodebuild:  /{{PATH}}/Texture/Source/ASDisplayNode+Yoga.mm:99:13: note: enclose 'userInterfaceLayoutDirectionForSemanticContentAttribute:' in an @available check to silence this warning
    - WARN  | [Texture/Yoga] xcodebuild:  /{{PATH}}/Xcode/DerivedData/App-akhbasdbsjtjlsgxcsxwstypgecs/Build/Products/Release-iphonesimulator/Texture/AsyncDisplayKit.framework/Headers/ASDisplayNode+Beta.h:180:44: warning: 'UISemanticContentAttribute' is partial: introduced in iOS 9.0 [-Wunguarded-availability]
    - NOTE  | [Texture/Yoga] xcodebuild:  /{{PATH}}/Xcode/DerivedData/App-akhbasdbsjtjlsgxcsxwstypgecs/Build/Products/Release-iphonesimulator/Texture/AsyncDisplayKit.framework/Headers/ASDisplayNode+Beta.h:180:1: note: annotate 'semanticContentAttributeDidChange:' with an availability attribute to silence
```

Also i opened [PR](https://github.com/pinterest/PINRemoteImage/pull/428) for PINRemoteImage.

UPD:
Here is the [log](https://github.com/TextureGroup/Texture/files/1474062/Texture.txt) of current master branch warnings

### CODE REVIEW

File: Tests/ASDisplayLayerTests.m [hunk 1]
```diff
@@ -472,7 +472,7 @@ - (void)DISABLED_testTransaction
   layer1.displaysAsynchronously = YES;
 
   dispatch_semaphore_t displayAsyncLayer1Sema = dispatch_semaphore_create(0);
-  layer1Delegate.displayLayerBlock = ^(_ASDisplayLayer *asyncLayer) {
+  layer1Delegate.displayLayerBlock = ^UIImage *{
```
Comment (Eke) [inline Tests/ASDisplayLayerTests.m]: Not sure about this one, tests are green but..

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by Kaspik** [comment]
👍 for this PR, currently warnings from Texture like `This block declaration is not a prototype` are the only ones in our project.

File: Source/Base/ASLog.h [hunk 2]
```diff
@@ -119,11 +119,44 @@ ASDISPLAYNODE_EXTERN_C_END
...
+
+#define as_log_debug(log, format, ...)            \
+if (AS_AVAILABLE_IOS(9)) {                        \
+  os_log_debug(log, format, ##__VA_ARGS__);       \
+} else {                                          \
+  (void)0;                                        \
+}                                                 \
+
+#define as_log_info(log, format, ...)             \
+if (AS_AVAILABLE_IOS(9)) { \
```
Comment (garrettmoon) [inline Source/Base/ASLog.h]: nit: indent `\`

File: Source/Base/ASLog.h [hunk 3]
```diff
@@ -119,11 +119,44 @@ ASDISPLAYNODE_EXTERN_C_END
...
+  os_log_info(log, format, ##__VA_ARGS__);        \
+} else {                                          \
+  (void)0;                                        \
+}\
```
Comment (garrettmoon) [inline Source/Base/ASLog.h]: same here

File: Source/Base/ASLog.h [hunk 4]
```diff
@@ -119,11 +119,44 @@ ASDISPLAYNODE_EXTERN_C_END
...
  * only available on iOS >= 9 at runtime, so just make them conditional.
  */
-#define as_log_create(subsystem, category)  (AS_AT_LEAST_IOS9 ? os_log_create(subsystem, category) : (os_log_t)0)
-#define as_log_debug(log, format, ...)      (AS_AT_LEAST_IOS9 ? os_log_debug(log, format, ##__VA_ARGS__) : (void)0)
-#define as_log_info(log, format, ...)       (AS_AT_LEAST_IOS9 ? os_log_info(log, format, ##__VA_ARGS__) : (void)0)
-#define as_log_error(log, format, ...)      (AS_AT_LEAST_IOS9 ? os_log_error(log, format, ##__VA_ARGS__) : (void)0)
-#define as_log_fault(log, format, ...)      (AS_AT_LEAST_IOS9 ? os_log_fault(log, format, ##__VA_ARGS__) : (void)0)
+
+#define as_log_create(subsystem, category) ({     \
+os_log_t __val;                                   \
```
Comment (garrettmoon) [inline Source/Base/ASLog.h]: Do the rest of the macros need to pass through their return value as well?

File: Tests/ASDisplayLayerTests.m (hunk 1 above)
Comment (garrettmoon) [reply Tests/ASDisplayLayerTests.m]: Looks right to me :)

File: Source/Base/ASLog.h (hunk 4 above)
Comment (Eke) [reply Source/Base/ASLog.h]: If i did not miss anything, other macros don't need to pass through their return value as well.

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/2368f05c2d3b473c2a87fa5b9d175bb1/raw/8f78a4d514f868f4ea8602041d18ef7604a4641c/log.txt)

**Comment by Eke** [comment]
Hello @garrettmoon, thank you for review. I resolved your comments. Tests are green on my local machine, i have no idea why they failed with CI.

**Comment by Kaspik** [comment]
@Eke Merge master in to restart the test and see if we can make this PR more up-to-date. :)

**Comment by nguyenhuy** [comment]
Alright, all green! Merging per @garrettmoon's review. Thank you all!

**Pull request merged**